File: android/app/src/main/java/com/ivpadim/stormy/CurrentWeather.java
package com.ivpadim.stormy;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

/**
 * Created by ivpadim on 4/6/15.
 */
public class CurrentWeather {
    private String mIcon;
    private long mTime;
    private double mTemperature;
    private double mHumidity;
    private double mPrecipitate;
    private String mSummary;
    private String mTimeZone;

    public String getTimeZone() {
        return mTimeZone;
    }

    public void setTimeZone(String timeZone) {
        mTimeZone = timeZone;
    }

    public String getIcon() {
        return mIcon;
    }

    public void setIcon(String icon) {
        this.mIcon = icon;
    }

    public int getIconId(){
        // clear-day, clear-night, rain, snow,
        // sleet, wind, fog, cloudy, partly-cloudy-day,
        // or partly-cloudy-night.
        int iconId = R.drawable.clear_day;

        switch (getIcon())
        {
            case "clear-day":
                iconId = R.drawable.clear_day;
                break;
            case "clear-night":
                iconId = R.drawable.clear_night;
                break;
            case "rain":
                iconId = R.drawable.rain;
                break;
            case "snow":
                iconId = R.drawable.snow;
                break;
            case "sleet":
                iconId = R.drawable.sleet;
                break;
            case "wind":
                iconId = R.drawable.wind;
                break;
            case "fog":
                iconId = R.drawable.fog;
                break;
            case "cloudy":
                iconId = R.drawable.cloudy;
                break;
            case "partly-cloudy-day":
                iconId = R.drawable.partly_cloudy;
                break;
            case "partly-cloudy-night":
                iconId = R.drawable.cloudy_night;
                break;
        }

        return iconId;
    }

    public long getTime() {
        return mTime;
    }

    public String getFormattedTime(){
        SimpleDateFormat formatter = new SimpleDateFormat("h:mm a");
        formatter.setTimeZone(TimeZone.getTimeZone(getTimeZone()));
        Date date = new Date(mTime * 1000);
        String timeString = formatter.format(date);

        return timeString;
    }

    public void setTime(long time) {
        this.mTime = time;
    }

    public int getTemperature() {
        return (int)Math.round(mTemperature);
    }

    public void setTemperature(double temperature) {
        this.mTemperature = temperature;
    }

    public double getHumidity() {
        return mHumidity;
    }

    public void setHumidity(double humidity) {
        this.mHumidity = humidity;
    }

    public int getPrecipitate() {
        double precipPercentage = mPrecipitate * 100;
        return (int)Math.round(precipPercentage);
    }

    public void setPrecipitate(double precipitate) {
        this.mPrecipitate = precipitate;
    }

    public String getSummary() {
        return mSummary;
    }

    public void setSummary(String summary) {
        this.mSummary = summary;
    }
}

File: ios/stormy/CurrentWeather.swift
//
//  CurrentWeather.swift
//  stormy
//
//  Created by ivpadim on 4/7/15.
//  Copyright (c) 2015 ivpadim. All rights reserved.
//

import Foundation
import UIKit

struct CurrentWeather{
    var time : Int //?
    var temperature : Int
    var humidity : Double
    var precipitate : Double
    var summary : String
    var icon : String
    
    init(weatherDictionary : NSDictionary){
        /*time = 1
        temperature = 10
        humidity = 0.0
        precipitate = 0.0
        summary = "no-summary"
        icon = "no-icon"*/
        let currentWeather  = weatherDictionary["currently"] as NSDictionary
        
        humidity = currentWeather["humidity"] as Double
        time = currentWeather["time"] as Int
        icon = currentWeather["icon"] as String;
        precipitate = currentWeather["precipProbability"] as Double
        summary = currentWeather["summary"] as String
        temperature = currentWeather["temperature"] as Int
    }
    
    func getFormattedTime() -> NSString {
        let timeInSeconds = NSTimeInterval(time)
        let date = NSDate(timeIntervalSince1970: timeInSeconds)
        
        let dateFormatter = NSDateFormatter()
        dateFormatter.timeStyle = .ShortStyle
        
        return dateFormatter.stringFromDate(date)
    }
    
    func getPrecipitate() -> Int {
        return Int(precipitate * 100)
    }
    
    func getIconImage() -> UIImage? {
        var iconImage : String
        
        switch icon
        {
        case "clear-day":
            iconImage = "clear-day"
        case "clear-night":
            iconImage="clear-night"
        case "rain":
            iconImage="rain"
        case "snow":
            iconImage="snow"
        case "sleet":
            iconImage="sleet"
        case "wind":
            iconImage="wind"
        case "fog":
            iconImage="fog"
        case "cloudy":
            iconImage="cloudy"
        case "partly-cloudy-day":
            iconImage="partly-cloudy"
        case "partly-cloudy-night":
            iconImage="cloudy-night"
        default:
            iconImage="default"
        }
        
        var image = UIImage(named: iconImage)
        
        return image
    }
}

